### PR TITLE
1182 validating on earliest and latest date

### PR DESCRIPTION
**Pull request by jasehumphr**
### What is the context of this PR?
Making sure that the user doesn't put an invalid validation in

### How to review 
Create a Date Answer and then add in an earlier Date then a Latest Date. 

You will have to run yarn on eq-author-api because I have added moment to the package.js

### CODE REVIEW

**Comment by heofs** [comment]
[Validating on Earliest and Latest Date (2)](https://trello.com/c/qDR6APvU/1182-validating-on-earliest-and-latest-date-2)

File: eq-author-api/schema/tests/validation.test.js [hunk 1]
```diff
@@ -414,6 +414,46 @@ describe("validation", () => {
...
+            custom: "2016-01-01",
+          },
+        });
+        const result = await queryValidation(ctx, answer.id);
+
+        expect(result.earliestDate.validationErrorInfo.errors).toMatchObject([
+          { errorCode: "ERR_EARLIEST_LATER_THAN_LATEST" },
+        ]);
+        expect(result.latestDate.validationErrorInfo.errors).toMatchObject([
+          { errorCode: "ERR_EARLIEST_LATER_THAN_LATEST" },
```
Comment (heofs) [inline eq-author-api/schema/tests/validation.test.js]: Maybe this errorCode should be pulled out into a constant

File: eq-author-api/src/validation/index.js [hunk 2]
```diff
@@ -44,6 +48,7 @@ module.exports = questionnaire => {
   //These are errors that are reported in two or more places however we only want to add to the total count once.
   const duplicatedErrorMessages = {
     ERR_MIN_LARGER_THAN_MAX: { occurrencesPerError: 2 },
+    ERR_EARLIEST_LATER_THAN_LATEST: { occurrencesPerError: 2 },
```
Comment (heofs) [inline eq-author-api/src/validation/index.js]: This too in a constant.

File: eq-author-api/src/validation/index.js [hunk 3]
```diff
@@ -12,6 +12,8 @@ const {
   VALIDATION,
   MIN_VALUE,
   MAX_VALUE,
+  STARTDATE_VALUE,
+  ENDDATE_VALUE,
```
Comment (heofs) [inline eq-author-api/src/validation/index.js]: Can you change the these to END_DATE_VALUE?

File: eq-author-api/src/validation/index.test.js [hunk 4]
```diff
@@ -207,6 +208,112 @@ describe("schema validation", () => {
...
+          validation: {
+            minValue: {
+              id: "123",
+              enabled: entity === "earliestDate",
+              custom: "2019-06-23",
+              inclusive: true,
+              entityType: "Custom",
+              previousAnswer: null,
+            },
+            maxValue: {
```
Comment (heofs) [inline eq-author-api/src/validation/index.test.js]: Should these be earliest and latest instead?

File: eq-author-api/tests/utils/contextBuilder/validation/queryValidation.js [hunk 5]
```diff
@@ -105,6 +115,16 @@ const getValidationQuery = `
     availableMetadata {
       id
     }
+    validationErrorInfo {
```
Comment (heofs) [inline eq-author-api/tests/utils/contextBuilder/validation/queryValidation.js]: Move the two validationErrorInfo into a fragment

File: eq-author/src/graphql/fragments/latest-date-validation-rule.graphql [hunk 6]
```diff
@@ -16,4 +16,14 @@ fragment LatestDateValidationRule on LatestDateValidationRule {
     id
     displayName
   }
+  validationErrorInfo {
```
Comment (heofs) [inline eq-author/src/graphql/fragments/latest-date-validation-rule.graphql]: Is it a fragment for this?

File: eq-author/src/graphql/fragments/earliest-date-validation-rule.graphql [hunk 7]
```diff
@@ -16,4 +16,14 @@ fragment EarliestDateValidationRule on EarliestDateValidationRule {
     id
     displayName
   }
+  validationErrorInfo {
```
Comment (heofs) [inline eq-author/src/graphql/fragments/earliest-date-validation-rule.graphql]: Is it a fragment for this?

File: eq-author-api/src/validation/schemas/entities.json [hunk 8]
```diff
@@ -142,75 +142,152 @@
...
-        "type": "object",
-        "properties": {
-          "validation": {
-            "if": {
-              "allOf": [
-                {
-                  "properties": {
-                    "minValue": {
-                      "properties": { "enabled": { "const": true } }
+      "allOf": [
```
Comment (SamGodwin2) [inline eq-author-api/src/validation/schemas/entities.json]: Can we make this a dateAnswer validation type rather than a basic answer?

**Pull request merged**